drivers/i2c/i2c_mcux_lpi2c.c

```diff
@@ -16,6 +16,8 @@
 LOG_MODULE_REGISTER(mcux_lpi2c);
 
 #include "i2c-priv.h"
+#define DELAY_TIME(baudrate) \
+	(1e6 * 10 * 10000 * 150 / (baudrate / 1000) / 600000000)
```

Comment:
This looks like it uses the system clock frequency (600 MHz), will it work with other clock frequencies? You probably need to use `clock_control_get_rate` similar to `mcux_lpi2c_configure`.

Reply:
Yes; and the logic behind this calculation needs some explanation. Why is the numerator 1.5E12?

Reply:
Hi，Be honest,  i test many values, the 10000 * 150 / (baudrate / 1000) is an experience one. I do not know why too.  But it works. bigger is maybe ok, but lower, not.
The values: the 1e6 means -> us, 10 is a simmliar ticks with (for (i=0;i<xx;i++), this will expand to 10 instructs costs about 
10 ticks), the 600000000 is sure, the cpu frequency.

Reply:
> This looks like it uses the system clock frequency (600 MHz), will it work with other clock frequencies? You probably need to use `clock_control_get_rate` similar to `mcux_lpi2c_configure`.

Ok， I will update the code soon.

Reply:
> This looks like it uses the system clock frequency (600 MHz), will it work with other clock frequencies? You probably need to use `clock_control_get_rate` similar to `mcux_lpi2c_configure`.

Hi, I have updated the macro with a function to get the CPU_FREQ, But I use CLOCK_GetCoreSysClkFreq instead, because the clock_control_get_rate can not return the core_freq, or do we need to add this into the mcux_ccm_get_subsys_rate()?